src/participant/participant.ts

```diff
@@ -67,24 +74,61 @@ export function getRunnableContentFromString(text: string): string {
 
 export default class ParticipantController {
   _participant?: vscode.ChatParticipant;
+  _context: vscode.ExtensionContext;
   _connectionController: ConnectionController;
   _storageController: StorageController;
   _chatMetadataStore: ChatMetadataStore;
+  _docsChatbotAIService: DocsChatbotAIService;
   _telemetryService: TelemetryService;
 
   constructor({
+    context,
     connectionController,
     storageController,
     telemetryService,
   }: {
+    context: vscode.ExtensionContext;
     connectionController: ConnectionController;
     storageController: StorageController;
     telemetryService: TelemetryService;
   }) {
+    this._context = context;
     this._connectionController = connectionController;
     this._storageController = storageController;
     this._chatMetadataStore = new ChatMetadataStore();
     this._telemetryService = telemetryService;
+
+    const docsChatbotBaseUri = this._readDocsChatbotBaseUri();
+    this._docsChatbotAIService = new DocsChatbotAIService(docsChatbotBaseUri);
+  }
+
+  // To integrate with the MongoDB documentation chatbot,
+  // set the MONGODB_DOCS_CHATBOT_BASE_URI environment variable when running the extension from a branch.
+  // This variable is automatically injected during the .vsix build process via GitHub Actions.
+  private _readDocsChatbotBaseUri(): string | undefined {
+    config({ path: path.join(this._context.extensionPath, '.env') });
+
+    try {
+      const docsChatbotBaseUriFileLocation = path.join(
+        this._context.extensionPath,
+        './constants.json'
+      );
+      // eslint-disable-next-line no-sync
+      const constantsFile = fs.readFileSync(
```

Comment:
Should we use the await syntax for this? 
```ts
import { promises as fs } from 'fs';
...
const constantsFile = await fs.readFile(
```
We could then, in the constructor, attach the response `_readDocsChatbotBaseUri` function to a promise, which we then await in the docs command handler. So it happens behind the scenes.